### PR TITLE
FIX plot_compared_evokeds grad ci estimation

### LINKED ISSUE #4730: Compare evokeds confidence intervals for grads
Still not clear to me if there is a bug in how confidence intervals for gradiometers in `plot_compare_evokeds` are computed? Using simulation [example](http://www.nmr.mgh.harvard.edu/mne/dev/auto_examples/simulation/plot_simulate_evoked_data.html#sphx-glr-auto-examples-simulation-plot-simulate-evoked-data-py) to illustrate...

screen shot 2017-11-04 at 8 57 31 pm

```
import numpy as np
import matplotlib.pyplot as plt

import mne
from mne.datasets import sample
from mne.time_frequency import fit_iir_model_raw
from mne.viz import plot_sparse_source_estimates
from mne.simulation import simulate_sparse_stc, simulate_evoked
data_path = sample.data_path()

raw = mne.io.read_raw_fif(data_path + '/MEG/sample/sample_audvis_raw.fif')
proj = mne.read_proj(data_path + '/MEG/sample/sample_audvis_ecg-proj.fif')
raw.info['projs'] += proj

fwd = mne.read_forward_solution(data_path + 
                                '/MEG/sample/sample_audvis-meg-eeg-oct-6-fwd.fif')
fwd = mne.pick_types_forward(fwd, meg=True, eeg=True, exclude=[])
cov = mne.read_cov(data_path + 
                   '/MEG/sample/sample_audvis-cov.fif')
info = mne.io.read_info(data_path + 
                        '/MEG/sample/sample_audvis-no-filter-ave.fif')

label_names = ['Aud-lh', 'Aud-rh']
labels = [mne.read_label(data_path + '/MEG/sample/labels/%s.label' % ln)
          for ln in label_names]
times = np.arange(300, dtype=np.float) / raw.info['sfreq'] - 0.1
picks = mne.pick_types(raw.info, meg=True, exclude='bads')
iir_filter = fit_iir_model_raw(raw, order=5, picks=picks, tmin=60, tmax=180)[1]
nave = 100  # simulate average of 100 epoc
rng = np.random.RandomState(42)


def data_fun(times):
    """Function to generate random source time courses"""
    return (50e-9 * np.sin(30. * times) *
            np.exp(- (times - 0.15 + 0.05 * rng.randn(1)) ** 2 / 0.01))

evokeds = list()
for i in range(10):
    stc = simulate_sparse_stc(fwd['src'], n_dipoles=2, times=times,
                              random_state=42, labels=labels, data_fun=data_fun)
    evokeds.append(simulate_evoked(fwd, stc, info, cov, nave=nave, use_cps=True,
                                   iir_filter=iir_filter))
picks = [raw.ch_names.index(j) for j in ['MEG 1811',
                                          'MEG 1812',
                                          'MEG 1813']]
mne.viz.plot_compare_evokeds(evokeds=[evokeds], picks=picks,
                             ci=.95)
```

### PR DESCRIPTION
**Pull request by ktavabi**
Attempt to fix #4730

### CODE REVIEW

File: mne/viz/evoked.py [hunk 1]
```diff
@@ -1664,8 +1664,13 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
             for condition in conditions:
                 # this will fail if evokeds do not have the same structure
                 # (e.g. channel count)
-                data = np.asarray([evoked_.data[picks, :].mean(0)
-                                   for evoked_ in evokeds[condition]])
+                if ch_type == 'grad':
+                    data = [(_merge_grad_data(evoked_.data[picks, :])).
```
Comment (jona-sassenhagen) [inline mne/viz/evoked.py]: It seems you need to import `_merge_grad_data` :)

Also maybe you will have to include a check that the relevant channels are in the picks.

**Comment by jona-sassenhagen** [comment]
Thanks for tackling this!

**Comment by ktavabi** [comment]
Shouldn't there be an import error if `_merge_grad_data` is not available?  In any case, I tried `from ..channels.layout import _merge_grad_data` at ln 1668 to no avail. 

I also tried unpacking and running `[(_merge_grad_data(evoked_.data[picks, :])).mean(0) for evoked_ in evokeds[condition]]` line by line with same result. 

> Also maybe you will have to include a check that the relevant channels are in the picks.

I am also suspecting the pairing of grads is somehow getting botched between ci and averaging, but can't see how. Anyway TBC, late here.

**Comment by jona-sassenhagen** [comment]
>Shouldn't there be an import error if _merge_grad_data is not available?

No, if you try to use something that's not defined, you get a NameError. ImportError is for stuff that you try to import, but the import fails.

**Comment by ktavabi** [comment]
Sorry, that's what I meant. I think `_merge_grad_data` is already in the namespace.

**Comment by ktavabi** [comment]
So it looks like 

1. Taking RMS of grad pairs
2. Computing ci and mean

is different from
1. Taking RMS of grads
2. computing ci
3. averaging evokeds
4. Taking RMS of grand mean

Does this make sense?

**Comment by agramfort** [comment]
@ktavabi to compute CIs on combined grads you need first to combine grads
on each epoch, then compute CIs and show the average + CIs.

in other words you cannot compute CIs on each channel 

clear?

**Comment by jona-sassenhagen** [comment]
Import of `merge_grads` depends on `if ch_type == 'grad' and len(picks) > 1:`. So if you have only one pick, it's not being imported.  However, if there is only one pick, you can't pair them anyways.

**Comment by ktavabi** [comment]
Sorry, @agramfort I am not sure what you mean.

 Where `ch_type == 'grad' and len(picks) > 1`

1. `data = np.asarray([(_merge_grad_data(evoked_.data[picks, :])).mean(0) for evoked_ in evokeds[condition]])` is the merged grad data array (cond x time) for each evoked ds?
2.  `data.mean(0)` and `_ci_fun(data)` are grad RMS mean and ci across evokeds?

I was assuming that is the scope of this PR i.e., we need mean + CIs of grad RMS (fig 2) across evokeds (fig 3) for gist in #4730 to work, no? 

screen shot 2017-11-10 at 11 48 33 am

**Comment by jona-sassenhagen** [comment]
Sorry, I can't be of much help here. I messed it up in the first place, grads just confuse me.

**Pull request closed without merging**